examples/five_mem_node/main.rs

```diff
@@ -169,7 +169,7 @@ impl Node {
         }
         let mut cfg = example_config();
         cfg.id = msg.get_to();
-        let storage = MemStorage::new();
+        let storage = MemStorage::new_with_conf_state(ConfState::from((vec![], vec![])));
```

Comment:
Here we can't iniitalize followers with a configuration because if we do so, the follower can't retrieve correct configuration from its leader.
Suppose the leader's id is `1` and it's initialized with configuration `[1]`, and after it becomes leader,  propose a configuration change to add peer `2`. Now the leader should have raft logs:
```
Empty | InitialConf(vec![1],vec![]) | Empty | AddNode(2)
```
There are 4 entries, and their indices are from `0` to `3`.
If the follower is initialized with any configuration, its raft logs will be:
```
Empty | AnyInitialConf
```
Indices from `0` to `1`. So the leader will send `MsgAppend` with entries `Empty | AddNode(2)` to the follower, which will cause an incorrect configuration on the follower.
So, if followers need to be initialized from leader's messages, it can't be initialized with any configuration.

Reply:
The example panics because of a bug introduced by #196 , sorry for that and I will send a PR to fix it, please help me to review that... By the way you can keep the PR to fix the another issue: the example won't exit because it's blocked on waiting all child threads. :(

Reply:
Yes, I think we can make this a fairly short running example. :)